### DISCUSSION TITLE
API function to exit out of all currently active snippets?

### DISCUSSION THREAD

**Discussion by nikfp** [L3MON4D3/LuaSnip#907] [Q&A] [answered]
I'm running into an issue with node jump behavior when I exit insert mode and move elsewhere, where the NEXT time I try to jump in nodes it randomly moves the cursor. I assume this is because I never completed a previous snip and Luasnip is trying to continue it. 

So question: What in the Luasnip API is the appropriate method to exit all currently running snips? I can't pin down which to use, but once I do I can use it to create an autocmd bound to InsertLeave.

**Comment by L3MON4D3** [comment, marked as answer]
You can do `ls.session.current_nodes[vim.api.nvim_get_current_buf()] = nil`, that will remove the currrent node for the current buffer (ie. you won't have history)

**Reply by nikfp** [reply]
Thank you so much for taking the time to reply to this, I'll integrate this into my config right now. Much appreciated!
